Build failure in `libspdm_rsp_challenge_auth.c`

The combination of `LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP=1` and `LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP=0` causes a build failure.
```
...\libspdm_rsp_challenge_auth.c(224): error C2220: the following warning is treated as an error
...\libspdm_rsp_challenge_auth.c(224): warning C4013: 'libspdm_generate_measurement_summary_hash' undefined; assuming extern returning int
```

--- a/library/spdm_responder_lib/libspdm_rsp_challenge_auth.c
+++ b/library/spdm_responder_lib/libspdm_rsp_challenge_auth.c
@@ -218,9 +218,9 @@ libspdm_return_t libspdm_get_response_challenge_auth(libspdm_context_t *spdm_con
 
     measurement_summary_hash = ptr;
 
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     if (libspdm_is_capabilities_flag_supported(
             spdm_context, false, 0, SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP)) {
-
         result = libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
             spdm_context,
@@ -241,6 +241,8 @@ libspdm_return_t libspdm_get_response_challenge_auth(libspdm_context_t *spdm_con
                                                SPDM_ERROR_CODE_UNSPECIFIED, 0,
                                                response_size, response);
     }
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
+
     ptr += measurement_summary_hash_size;
 
     opaque_data_size = *response_size - (sizeof(spdm_challenge_auth_response_t) + hash_size +

--- a/library/spdm_responder_lib/libspdm_rsp_key_exchange.c
+++ b/library/spdm_responder_lib/libspdm_rsp_key_exchange.c
@@ -501,6 +501,7 @@ libspdm_return_t libspdm_get_response_key_exchange(libspdm_context_t *spdm_conte
 
     ptr += dhe_key_size;
 
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     if (libspdm_is_capabilities_flag_supported(
             spdm_context, false, 0, SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP)) {
 
@@ -525,6 +526,8 @@ libspdm_return_t libspdm_get_response_key_exchange(libspdm_context_t *spdm_conte
                                                SPDM_ERROR_CODE_UNSPECIFIED, 0,
                                                response_size, response);
     }
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
+
     ptr += measurement_summary_hash_size;
 
     libspdm_write_uint16(ptr, (uint16_t)opaque_key_exchange_rsp_size);

--- a/library/spdm_responder_lib/libspdm_rsp_psk_exchange.c
+++ b/library/spdm_responder_lib/libspdm_rsp_psk_exchange.c
@@ -319,6 +319,7 @@ libspdm_return_t libspdm_get_response_psk_exchange(libspdm_context_t *spdm_conte
 
     ptr = (void *)(spdm_response + 1);
 
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     if (libspdm_is_capabilities_flag_supported(
             spdm_context, false, 0,  SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP)) {
 
@@ -343,6 +344,8 @@ libspdm_return_t libspdm_get_response_psk_exchange(libspdm_context_t *spdm_conte
                                                SPDM_ERROR_CODE_UNSPECIFIED, 0,
                                                response_size, response);
     }
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
+
     ptr += measurement_summary_hash_size;
 
     if (context_length != 0) {

--- a/os_stub/spdm_device_secret_lib_null/lib.c
+++ b/os_stub/spdm_device_secret_lib_null/lib.c
@@ -50,34 +50,6 @@ bool libspdm_measurement_opaque_data(
     return false;
 }
 
-bool libspdm_challenge_opaque_data(
-#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
-    void *spdm_context,
-#endif
-    spdm_version_number_t spdm_version,
-    uint8_t slot_id,
-    uint8_t *measurement_summary_hash,
-    size_t measurement_summary_hash_size,
-    void *opaque_data,
-    size_t *opaque_data_size)
-{
-    return false;
-}
-
-bool libspdm_encap_challenge_opaque_data(
-#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
-    void *spdm_context,
-#endif
-    spdm_version_number_t spdm_version,
-    uint8_t slot_id,
-    uint8_t *measurement_summary_hash,
-    size_t measurement_summary_hash_size,
-    void *opaque_data,
-    size_t *opaque_data_size)
-{
-    return false;
-}
-
 bool libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
     void *spdm_context,
@@ -93,6 +65,38 @@ bool libspdm_generate_measurement_summary_hash(
     return false;
 }
 #endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
+
+#if LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP
+bool libspdm_challenge_opaque_data(
+#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
+    void *spdm_context,
+#endif
+    spdm_version_number_t spdm_version,
+    uint8_t slot_id,
+    uint8_t *measurement_summary_hash,
+    size_t measurement_summary_hash_size,
+    void *opaque_data,
+    size_t *opaque_data_size)
+{
+    return false;
+}
+#endif /* LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP */
+
+#if LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP
+bool libspdm_encap_challenge_opaque_data(
+#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
+    void *spdm_context,
+#endif
+    spdm_version_number_t spdm_version,
+    uint8_t slot_id,
+    uint8_t *measurement_summary_hash,
+    size_t measurement_summary_hash_size,
+    void *opaque_data,
+    size_t *opaque_data_size)
+{
+    return false;
+}
+#endif /* LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP */
 
 #if LIBSPDM_ENABLE_CAPABILITY_MEL_CAP
 /*Collect the measurement extension log.*/

--- a/os_stub/spdm_device_secret_lib_sample/lib.c
+++ b/os_stub/spdm_device_secret_lib_sample/lib.c
@@ -1634,58 +1634,6 @@ bool libspdm_measurement_opaque_data(
     return true;
 }
 
-size_t libspdm_secret_lib_challenge_opaque_data_size;
-
-bool libspdm_challenge_opaque_data(
-#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
-    void *spdm_context,
-#endif
-    spdm_version_number_t spdm_version,
-    uint8_t slot_id,
-    uint8_t *measurement_summary_hash,
-    size_t measurement_summary_hash_size,
-    void *opaque_data,
-    size_t *opaque_data_size)
-{
-    size_t index;
-
-    LIBSPDM_ASSERT(libspdm_secret_lib_challenge_opaque_data_size <= *opaque_data_size);
-
-    *opaque_data_size = libspdm_secret_lib_challenge_opaque_data_size;
-
-    for (index = 0; index < *opaque_data_size; index++)
-    {
-        ((uint8_t *)opaque_data)[index] = (uint8_t)index;
-    }
-
-    return true;
-}
-
-bool libspdm_encap_challenge_opaque_data(
-#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
-    void *spdm_context,
-#endif
-    spdm_version_number_t spdm_version,
-    uint8_t slot_id,
-    uint8_t *measurement_summary_hash,
-    size_t measurement_summary_hash_size,
-    void *opaque_data,
-    size_t *opaque_data_size)
-{
-    size_t index;
-
-    LIBSPDM_ASSERT(libspdm_secret_lib_challenge_opaque_data_size <= *opaque_data_size);
-
-    *opaque_data_size = libspdm_secret_lib_challenge_opaque_data_size;
-
-    for (index = 0; index < *opaque_data_size; index++)
-    {
-        ((uint8_t *)opaque_data)[index] = (uint8_t)index;
-    }
-
-    return true;
-}
-
 bool libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
     void *spdm_context,
@@ -1810,8 +1758,63 @@ bool libspdm_generate_measurement_summary_hash(
     }
     return true;
 }
-
 #endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
+
+#if LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP
+size_t libspdm_secret_lib_challenge_opaque_data_size;
+
+bool libspdm_challenge_opaque_data(
+#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
+    void *spdm_context,
+#endif
+    spdm_version_number_t spdm_version,
+    uint8_t slot_id,
+    uint8_t *measurement_summary_hash,
+    size_t measurement_summary_hash_size,
+    void *opaque_data,
+    size_t *opaque_data_size)
+{
+    size_t index;
+
+    LIBSPDM_ASSERT(libspdm_secret_lib_challenge_opaque_data_size <= *opaque_data_size);
+
+    *opaque_data_size = libspdm_secret_lib_challenge_opaque_data_size;
+
+    for (index = 0; index < *opaque_data_size; index++)
+    {
+        ((uint8_t *)opaque_data)[index] = (uint8_t)index;
+    }
+
+    return true;
+}
+#endif /* LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP */
+
+#if LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP
+bool libspdm_encap_challenge_opaque_data(
+#if LIBSPDM_HAL_PASS_SPDM_CONTEXT
+    void *spdm_context,
+#endif
+    spdm_version_number_t spdm_version,
+    uint8_t slot_id,
+    uint8_t *measurement_summary_hash,
+    size_t measurement_summary_hash_size,
+    void *opaque_data,
+    size_t *opaque_data_size)
+{
+    size_t index;
+
+    LIBSPDM_ASSERT(libspdm_secret_lib_challenge_opaque_data_size <= *opaque_data_size);
+
+    *opaque_data_size = libspdm_secret_lib_challenge_opaque_data_size;
+
+    for (index = 0; index < *opaque_data_size; index++)
+    {
+        ((uint8_t *)opaque_data)[index] = (uint8_t)index;
+    }
+
+    return true;
+}
+#endif /* LIBSPDM_ENABLE_CAPABILITY_CHAL_CAP */
 
 #if LIBSPDM_ENABLE_CAPABILITY_MEL_CAP
 /*Collect the measurement extension log.*/

--- a/unit_test/test_spdm_responder/key_exchange.c
+++ b/unit_test/test_spdm_responder/key_exchange.c
@@ -707,7 +707,9 @@ void libspdm_test_responder_key_exchange_case8(void **state)
     libspdm_context_t *spdm_context;
     size_t response_size;
     uint8_t response[LIBSPDM_MAX_SPDM_MSG_SIZE];
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     uint8_t measurement_hash[LIBSPDM_MAX_HASH_SIZE];
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
     uint32_t measurement_summary_hash_size;
     spdm_key_exchange_response_t *spdm_response;
     void *data1;
@@ -792,6 +794,7 @@ void libspdm_test_responder_key_exchange_case8(void **state)
 
     measurement_summary_hash_size = libspdm_get_measurement_summary_hash_size(
         spdm_context, false, m_libspdm_key_exchange_request3.header.param1);
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
         spdm_context,
@@ -810,6 +813,7 @@ void libspdm_test_responder_key_exchange_case8(void **state)
         dhe_key_size,
         measurement_hash,
         measurement_summary_hash_size);
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
     free(data1);
 }
 
@@ -820,7 +824,9 @@ void libspdm_test_responder_key_exchange_case9(void **state)
     libspdm_context_t *spdm_context;
     size_t response_size;
     uint8_t response[LIBSPDM_MAX_SPDM_MSG_SIZE];
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     uint8_t measurement_hash[LIBSPDM_MAX_HASH_SIZE];
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
     uint32_t measurement_summary_hash_size;
     spdm_key_exchange_response_t *spdm_response;
     void *data1;
@@ -906,6 +912,7 @@ void libspdm_test_responder_key_exchange_case9(void **state)
 
     measurement_summary_hash_size = libspdm_get_measurement_summary_hash_size(
         spdm_context, false, m_libspdm_key_exchange_request4.header.param1);
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
         spdm_context,
@@ -924,6 +931,7 @@ void libspdm_test_responder_key_exchange_case9(void **state)
         dhe_key_size,
         measurement_hash,
         measurement_summary_hash_size);
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
 
     session_id = (m_libspdm_key_exchange_request4.req_session_id << 16) |
                  spdm_response->rsp_session_id;

--- a/unit_test/test_spdm_responder/psk_exchange.c
+++ b/unit_test/test_spdm_responder/psk_exchange.c
@@ -913,7 +913,9 @@ void libspdm_test_responder_psk_exchange_case10(void **state)
     libspdm_context_t *spdm_context;
     size_t response_size;
     uint8_t response[LIBSPDM_MAX_SPDM_MSG_SIZE];
+    #if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     uint8_t measurement_hash[LIBSPDM_MAX_HASH_SIZE];
+    #endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
     uint32_t measurement_summary_hash_size;
     spdm_psk_exchange_response_t *spdm_response;
     void *data1;
@@ -1002,6 +1004,8 @@ void libspdm_test_responder_psk_exchange_case10(void **state)
 
     measurement_summary_hash_size = libspdm_get_measurement_summary_hash_size(
         spdm_context, false, m_libspdm_psk_exchange_request4.header.param1);
+
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
         spdm_context,
@@ -1013,11 +1017,13 @@ void libspdm_test_responder_psk_exchange_case10(void **state)
         m_libspdm_psk_exchange_request4.header.param1,
         measurement_hash,
         measurement_summary_hash_size);
+
     assert_memory_equal(
         (uint8_t *)response +
         sizeof(spdm_psk_exchange_response_t),
         measurement_hash,
         measurement_summary_hash_size);
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
 
 #if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
     assert_int_equal(spdm_context->transcript.message_m.buffer_size,
@@ -1033,7 +1039,9 @@ void libspdm_test_responder_psk_exchange_case11(void **state)
     libspdm_context_t *spdm_context;
     size_t response_size;
     uint8_t response[LIBSPDM_MAX_SPDM_MSG_SIZE];
+    #if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     uint8_t measurement_hash[LIBSPDM_MAX_HASH_SIZE];
+    #endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
     uint32_t measurement_summary_hash_size;
     spdm_psk_exchange_response_t *spdm_response;
     void *data1;
@@ -1122,6 +1130,8 @@ void libspdm_test_responder_psk_exchange_case11(void **state)
 
     measurement_summary_hash_size = libspdm_get_measurement_summary_hash_size(
         spdm_context, false, m_libspdm_psk_exchange_request5.header.param1);
+
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     libspdm_generate_measurement_summary_hash(
 #if LIBSPDM_HAL_PASS_SPDM_CONTEXT
         spdm_context,
@@ -1133,11 +1143,13 @@ void libspdm_test_responder_psk_exchange_case11(void **state)
         m_libspdm_psk_exchange_request5.header.param1,
         measurement_hash,
         measurement_summary_hash_size);
+
     assert_memory_equal(
         (uint8_t *)response +
         sizeof(spdm_psk_exchange_response_t),
         measurement_hash,
         measurement_summary_hash_size);
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
 
 #if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
     assert_int_equal(spdm_context->transcript.message_m.buffer_size,

--- a/unit_test/test_spdm_responder/receive_send.c
+++ b/unit_test/test_spdm_responder/receive_send.c
@@ -67,6 +67,7 @@ libspdm_return_t my_test_get_response_func2(
  **/
 void libspdm_test_responder_receive_send_rsp_case1(void** state)
 {
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     /* This test case is partially copied from test_requester_get_measurement_case4 */
     libspdm_return_t status;
     libspdm_test_context_t* spdm_test_context;
@@ -191,6 +192,7 @@ void libspdm_test_responder_receive_send_rsp_case1(void** state)
     libspdm_asym_free(spdm_context->connection_info.algorithm.base_asym_algo,
                       spdm_context->connection_info.peer_used_cert_chain[0].leaf_cert_public_key);
     #endif
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
 }
 
 /**
@@ -363,6 +365,7 @@ void libspdm_test_responder_receive_send_rsp_case3(void** state)
  **/
 void libspdm_test_responder_receive_send_rsp_case4(void** state)
 {
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
     /* This test case is partially copied from test_requester_get_measurement_case4 */
     libspdm_return_t status;
     libspdm_test_context_t* spdm_test_context;
@@ -485,7 +488,9 @@ void libspdm_test_responder_receive_send_rsp_case4(void** state)
     libspdm_asym_free(spdm_context->connection_info.algorithm.base_asym_algo,
                       spdm_context->connection_info.peer_used_cert_chain[0].leaf_cert_public_key);
     #endif
+#endif /* LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP */
 }
+
 
 libspdm_test_context_t m_libspdm_responder_receive_send_test_context = {
     LIBSPDM_TEST_CONTEXT_VERSION,

